napari/utils/validators.py

```diff
@@ -82,3 +84,34 @@ def func(obj):
                     )
 
     return func
+
+
+def pairwise(iterable: Iterable):
```

Comment:
I'm not sure if this was part of the discussion around a previous version of this PR, but this feel like a premature refactor.

I think the only place this `validators` module is used is in the `intensity_mixin` module. Maybe it could be used elsewhere in the future, but right now I think we have defined 4 functions just to validate that `contrast_limits` and `contrast_limits_range` is a pair of increasing values.

Rather than expand the public API of napari and make the call sequence deeper, I'd suggest defining a private help function like

```python
def _is_increasing_pair(value: Any) -> bool:
    return isinstance(value, Sequence) and len(value) == 2 and value[0] < value[1]
```

And then use it in the setters like

```python
if not _is_increasing_pair(contrast_limits):
    raise ValueError(f'contrast_limits should be a pair of increasing values. Instead got {contrast_llimits}.')
```

We could define a function that raises instead, but then we have to pass the name of the attribute through, which hurts readability. You could also do a checked conversion to `Tuple` if we want to support generators, but that doesn't seem particularly important to me.

Despite the length of my comment here, this isn't blocking - I'm just relatively confused how we got here.

Reply:
This was indeed part of a debate in the previous version of this PR (https://github.com/napari/napari/pull/4159#discussion_r815508225)

Reply:
OK, given the amount of previous discussion here, I'll let this go in.